azure/services/agentpools/spec.go

```diff
@@ -42,28 +41,34 @@ type KubeletConfig struct {
 	// CPUCfsQuotaPeriod - Sets CPU CFS quota period value.
 	CPUCfsQuotaPeriod *string
 	// ImageGcHighThreshold - The percent of disk usage after which image garbage collection is always run.
-	ImageGcHighThreshold *int32
+	ImageGcHighThreshold *int
```

Comment:
Lots of fields that were `*int32` in the SDK are now `*int` in ASO (or similar for other types). I elected not to change the API types but to change them everywhere else to match ASO.

Reply:
I think that's fair, changing the API type would be quite inconvenient as it would require adding new conversion functions for each `int32`.

Reply:
I asked in Slack if changing these types would be too disruptive and haven't gotten any feedback yet indicating they would be, so I'm still considering changing these to match ASO so CAPZ doesn't need to do any conversion.